### DISCUSSION TITLE
Enhancement Request: Add option to `isEmpty` to consider whitespace as empty

### DISCUSSION THREAD

**Discussion by indi-rin** [ueberdosis/tiptap#5413] [Feature Requests]
### Description

It would be helpful if there was a way to have `editor.isEmpty` return true if the editor contains only whitespaces. Maybe something like `editor.isEmpty({ ignoreWhitespace: true })` to support considering whitespace empty in addition to the existing behavior.

Currently, we can't use `editor.isEmpty` and instead have to check and trim the content with `editor.getText() && editor().getText().trim().length > 0`. It's not a terrible workaround but it'd be nice if we were able to utilize the built-in feature!

### Use Case

I want to use this feature for cases where we process user input if there is text in the editor differently from if there is no text in the editor, and we don't consider whitespace to be text that can be processed.

(Also, while checking to see if this behavior was supported, I came across [this StackOverflow](https://stackoverflow.com/questions/76466963/tiptap-make-isempty-return-true-when-the-editor-only-contains-whitespaces-and-n) from a year ago of other people asking about the same thing.)

### Type

Other

**Comment by nperez0111** [comment]
I would probably not have it be on the editor instance, but maybe as part of `isNodeEmpty` (which is what backs editor.isEmpty anyway).

I would be willing to accept a PR that adds to the second parameter `ignoreWhitespace` (default false) to this function https://github.com/ueberdosis/tiptap/blob/develop/packages/core/src/helpers/isNodeEmpty.ts

**Reply by indi-rin** [reply]
Thank you. We might consider such a contribution.

**Reply by nperez0111** [reply]
I have a PR for this: https://github.com/ueberdosis/tiptap/pull/5446

**Reply by indi-rin** [reply]
Thank you so much, really appreciate it!

**Reply by nperez0111** [reply]
It was released with 2.5.9
